jetty-webapp/src/main/java/org/eclipse/jetty/webapp/CachingWebAppClassLoader.java

```diff
@@ -23,25 +23,35 @@
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 
+import org.eclipse.jetty.server.handler.ContextHandler;
 import org.eclipse.jetty.util.annotation.ManagedObject;
 import org.eclipse.jetty.util.annotation.ManagedOperation;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.util.log.Log;
 import org.eclipse.jetty.util.log.Logger;
 
 /**
- * A WebAppClassLoader that caches {@link #getResource(String)} results.
+ * A WebAppClassLoader that caches {@link #getResource(String)} and
+ * {@link #loadAsResource(String, boolean)} results.
  * Specifically this ClassLoader caches not found classes and resources,
  * which can greatly increase performance for applications that search
  * for resources.
  */
 @ManagedObject
 public class CachingWebAppClassLoader extends WebAppClassLoader
 {
+    static
+    {
+        registerAsParallelCapable();
+    }
+
     private static final Logger LOG = Log.getLogger(CachingWebAppClassLoader.class);
 
     private final Set<String> _notFound = ConcurrentHashMap.newKeySet();
     private final ConcurrentHashMap<String, URL> _cache = new ConcurrentHashMap<>();
```

Comment:
This map will contain urls of resources and classes. When I added new classCache here I wanted to split urls - one for resources and one for classes.
If I have single cache, it leads to such error in my app:
Caused by: java.lang.ClassNotFoundException: org.springframework.web.SpringServletContainerInitializer
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382) ~[na:1.8.0_241]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418) ~[na:1.8.0_241]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355) ~[na:1.8.0_241]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351) ~[na:1.8.0_241]
	at org.eclipse.jetty.webapp.WebAppClassLoader.loadClass(WebAppClassLoader.java:555)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351) ~[na:1.8.0_241]
	at org.eclipse.jetty.webapp.CachingWebAppClassLoader.loadClass(CachingWebAppClassLoader.java:101) 
	at org.eclipse.jetty.plus.annotation.ContainerInitializer.<init>(ContainerInitializer.java:64)